src/gefs_6h_ave_1mem.fd/gefs_6h_ave_1mem.f90

```diff
@@ -0,0 +1,532 @@
+      character(len=8) :: file_date
+!      character(len=5) :: ens_mem
+      integer :: unit, ifid,ifid1,nx,ny,iret,jret,i,k,ifh,nfi,maxgrd,ifd,ind,ens_id
+      character(len=3) ::sfh
+      character(len=2) ::sdy
+      character(len=2) ::smonth
+      character(len=4) ::syr
+      character(len=8) :: pabbrev
+      character(len=30) :: labbrev
+      character(len=30) :: labbrev_short
+      character(len=500) :: datapath = './'
+
+      real, allocatable :: var_save(:,:) ! (maxgrd,nfi)
+      real, allocatable :: var_save_acc(:) ! (maxgrd)
+      real, allocatable :: var_save_DSWRF(:) ! (maxgrd)
+      real, allocatable :: var_save_USWRF(:) ! (maxgrd)
+      real, allocatable :: apcp_6h(:) ! (maxgrd)
+      real, allocatable :: acpcp_6h(:) ! (maxgrd)
+      real, allocatable :: apcp_3h(:) ! (maxgrd)
+      real, allocatable :: acpcp_3h(:) ! (maxgrd)
+      real, allocatable :: dpcp(:) ! (maxgrd)
+
+
+      call GET_COMMAND_ARGUMENT(1, file_date)
+!      call get_environment_variable("ens_mem", ens_mem)
+
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!open files for output grib2 (f03 and f006)
+       outfile03=trim(datapath)//'gefs'//'.t00z.'//'pgrb2af003'
+       print *,outfile03
+       call baopenwa(300,outfile03,iret) ! for add more than 1 members
+       outfile006=trim(datapath)//'gefs'//'.t00z.'//'pgrb2af006'
+       print *,outfile006
+       call baopenwa(200,outfile006,iret) ! for add more than 1 members
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+
+      do ifid=1,nfield1
+      nfi=1
+      do ifh=003,006,3  !foercast hours
+
+       nfi=nfi+1
+       write(sfh,'(i3.3)') ifh
+
+!search in pgrb2a
+
+!        datafile(nfi)=trim(file_dir)//'/gefs.'//trim(file_date)//'/00/'//trim(ens_mem)//'.t00z.pgrb2af'//trim(sfh)
+!        if(ifh == 000 .or. ifh == 003 ) then 
+        if( ifh == 003 ) then 
+           datafile(nfi)=trim(datapath)//'gefs.t00z.master.grb2f'//trim(sfh) ! only control data avail
+        else
+           datafile(nfi)=trim(datapath)//'gefs.t00z.master.grb2f'//trim(sfh)
+        endif
+
+
+      unit=100
+       call BAOPENR(unit, datafile(nfi), iret)
+       if(iret /= 0 ) then
+        write(*,*) "there is no GEFS forecast",datafile(nfi)
+       end if
+       write(*,*) unit, datafile(nfi), iret
+
+       j = 0
+       jids=-9999
+       jids=-9999;jpdt=-9999; jgdt=-9999
+       jdisc=-1; jgdtn=-1
+
+       if (nfi /= 3 ) then
+          jpdtn = pdt%npdt(ifid)  !template version num. 
+          jpdt(1)  = pdt%icat(ifid)
+          jpdt(2)  = pdt%iprm(ifid)
+          jpdt(10) = pdt%iffs(ifid)
+          jdisc   = disc(ifid)
+       else
+!          ifid1=ifid + nfield1/2
+          jpdtn =11  !template version num.
+          jpdt(1)  = pdt%icat(ifid)
+          jpdt(2)  = pdt%iprm(ifid)
+          jpdt(10) = pdt%iffs(ifid)
+          jdisc   = disc(ifid)
+!!          if (ifid ==14 ) jpdt(10) = 10
+       endif
```

Comment:
Line 171 and 178 are the only conditional dependent here.  There others can be outside the if-else structure.